### PR TITLE
Add asset inventory AWS RDS fetcher

### PR DESCRIPTION
**Pull request by kubasobon**
### Summary of your changes

Adds an AWS RDS instance fetcher.

### Screenshot/Data

Screenshot 2024-05-13 at 12 18 39

image

### Related Issues

Closes https://github.com/elastic/security-team/issues/9083

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

no

### CODE REVIEW

File: internal/inventory/asset.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ type assetType string
 const (
 	TypeVirtualMachine assetType = "virtual-machine"
 	TypeObjectStorage  assetType = "object-storage"
+	TypeDatabase       assetType = "database"
```
Comment (romulets) [inline internal/inventory/asset.go:51]: Maybe as type it could be `relational-database`?

**Review by romulets — APPROVED** [review]

**Comment by oren-zohar** [comment]
Why is there `database` twice in the index name?
image

**Comment by oren-zohar** [comment]
It's also unclear to me what the convention is for the index name when using `-` and `_`, I thought `-` is saved for the namespace. @romulets

**Comment by kubasobon** [comment]
@oren-zohar The index name is constructed [here](https://github.com/elastic/cloudbeat/blob/main/internal/inventory/inventory.go#L121-L123)
```
func generateIndex(a Asset) string {
	return fmt.Sprintf("asset_inventory_%s_%s_%s_%s", a.Category, a.SubCategory, a.Type, a.SubType)
}
```

`Type` changed to `relational-database`, so it will be a bit different now. Subdivision is based on [this chart](https://whimsical.com/asset-inventory-TbkK5f7Dhk7WtNpfP3YS95).

**Pull request merged**